study1/week4/valid-palindrome-ii.java

```diff
@@ -0,0 +1,22 @@
+// https://leetcode.com/problems/valid-palindrome-ii/
+// Time Complexity: O(n), n: s.length()/2
+// Space Complexity: O(1)
+class Solution {
+    public boolean validPalindrome(String s) {
+        for (int begin = 0, end = s.length()-1; begin < end; ++begin, --end) {
+            if (s.charAt(begin) != s.charAt(end)) {
+                return compareCharacter(s, begin+1, end) || compareCharacter(s, begin, end-1);
+            }
+        }
+        return true;
+    }
+
+    // check the other characters
+    public boolean compareCharacter(final String s, int left, int right) {
+        while (left < right) {
```

Comment:
I like this loop iteration, it would have been better if you did the same for the loop in line 6. 🙂